pkg/kubectl/scheme/scheme.go

```diff
@@ -50,5 +51,5 @@ var Versions = []schema.GroupVersion{}
 
 // DefaultJSONEncoder returns a default encoder for our scheme
 func DefaultJSONEncoder() runtime.Encoder {
-	return Codecs.LegacyCodec(Versions...)
+	return unstructured.JSONFallbackEncoder{Codecs.LegacyCodec(Versions...)}
```

Comment:
Actually no. The codec tries to convert into one of the given versions and fails otherwise. But we have to support arbitrary versions for Unstructured. My wrapper makes this explicit. IMO it is not the task of the codec to handle unknown versions, but it must be explicit.